src/nvim/api/vim.c

```diff
@@ -164,6 +165,39 @@ Object vim_eval(String str, Error *err)
   return rv;
 }
 
+/// Call the given function with the given arguments stored in an array.
+///
+/// @param functionName The function to call
+/// @param args the functions arguments packed in an Array
+/// @param[out] err Details of an error that may have occurred
+/// @return The result of the function call if it was successful
+Object vim_function(String functionName, Array args, Error *err)
+  FUNC_ATTR_DEFERRED
+{
+  Object rv = OBJECT_INIT;
+  typval_T vim_args[args.size+1];
+
+  try_start();
+  // Convert the arguments in args from Object to typval_T values
+  size_t i = 0;
+  for (; i < args.size; i++) {
+    if (!object_to_vim(args.items[i], &vim_args[i], err)) {
+      break;
+    }
+  }
+
+  typval_T tmp;
+  if (!err->set && func_call((char_u*) functionName.data, vim_args, NULL, &tmp) == FAIL) {
```

Comment:
To clarify my earlier comment, you may need something like `char* name = xstrndup(functionName.data, functionName.size)` here.

Also, `rettv` would be a more idiomatic name for the return value.